go/client/json_api_common.go

```diff
@@ -158,37 +158,51 @@ type handler interface {
 	handle(ctx context.Context, c Call, w io.Writer) error
 }
 
-func (c *cmdAPI) runHandler(h handler) error {
-	r := io.Reader(os.Stdin)
-	if len(c.message) > 0 {
-		r = strings.NewReader(c.message)
-	} else if len(c.inputFile) > 0 {
-		f, err := os.Open(c.inputFile)
+func (c *cmdAPI) runHandler(h handler) (err error) {
+	var r io.Reader
+	var w io.Writer
+	defer func() {
+		if err != nil {
+			enc := json.NewEncoder(w)
```

Comment:
why not use `encodeErr` here?

Reply:
required a call argument which we don't have (and I didn't know if it made sense to pull the first call off the input if we got an error before we started parsing those)

Reply:
just pass a `Call{}` in

Reply:
ah, right. still getting used to golang typing